misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/ProgramAnalyzerNodeVisitor.java

```diff
@@ -375,1133 +196,165 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
 
-    @Override
-    public void visit(DoStatementNode doStatementNode) {
+        if (node.kind() == SyntaxKind.BRACED_EXPRESSION) {
+            BracedExpressionNode bracedExpressionNode = (BracedExpressionNode) node;
+            node = bracedExpressionNode.expression();
 
-    }
+            if (node.kind() == SyntaxKind.BINARY_EXPRESSION) {
+                if (((BinaryExpressionNode) node).rhsExpr().kind() == SyntaxKind.NUMERIC_LITERAL &&
+                        ((BinaryExpressionNode) node).lhsExpr().kind() == SyntaxKind.NUMERIC_LITERAL) {
+                    long rhsValue = Long.parseLong(((BinaryExpressionNode) node).rhsExpr().toString().trim());
+                    long lhsValue = Long.parseLong(((BinaryExpressionNode) node).lhsExpr().toString().trim());
+                    iterationsCount = rhsValue - lhsValue + 1;
+                }
+            }
+        }
 
-    @Override
-    public void visit(ClassDefinitionNode classDefinitionNode) {
+        ForStatementNode forStatementNode = new ForStatementNode(iterationsCount);
+        Node currentParentNode = this.currentNode;
 
-    }
+        Node forBodyNode = new Node();
+        this.currentNode = forBodyNode;
 
-    @Override
-    public void visit(ResourcePathParameterNode resourcePathParameterNode) {
+        visitSyntaxNode(forEachStatementNode.blockStatement());
+        forStatementNode.setForBody(forBodyNode.getNextNode());
 
-    }
+        this.currentNode = currentParentNode;
 
-    @Override
-    public void visit(RequiredExpressionNode requiredExpressionNode) {
+        if (forEachStatementNode.blockStatement() != null) {
+            this.setChildNode(forStatementNode);
+        }
 
+        visitSyntaxNode(forEachStatementNode);
     }
 
     @Override
-    public void visit(ErrorConstructorExpressionNode errorConstructorExpressionNode) {
-
-    }
+    public void visit(WhileStatementNode whileStatementNode) {
 
-    @Override
-    public void visit(ParameterizedTypeDescriptorNode parameterizedTypeDescriptorNode) {
+        ForStatementNode forStatementNode = new ForStatementNode(DEFAULT_LOOP_SIZE);
+        Node currentParentNode = this.currentNode;
 
-    }
+        Node forBodyNode = new Node();
+        this.currentNode = forBodyNode;
 
-    @Override
-    protected void visitSyntaxNode(io.ballerina.compiler.syntax.tree.Node node) {
+        visitSyntaxNode(whileStatementNode.whileBody());
+        forStatementNode.setForBody(forBodyNode.getNextNode());
 
-        super.visitSyntaxNode(node);
+        this.currentNode = currentParentNode;
+        if (forStatementNode.getForBody() != null) {
+            this.setChildNode(forStatementNode);
+        }
     }
 
     @Override
-    public int hashCode() {
+    public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
 
-        return super.hashCode();
+        resolveActionInvocation(remoteMethodCallActionNode);
+        visitSyntaxNode(remoteMethodCallActionNode);
```

Comment:
We can use the accept method here as well. Correct me if i'm wrong. Isn't this a recursive call. Is it necessary. Can this be a never ending loop?